bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java

```diff
@@ -330,6 +331,28 @@ private BigtableOptions resolveLegacyOptions(BigtableOptions options) throws IOE
     return options;
   }
 
+
+  /**
+   * Return options with legacy input options, if any, resolved into currently supported options.
+   */
+  public synchronized BigtableClusterName getClusterName() throws IOException {
+    if (options.getClusterName() == null) {
```

Comment:
That is where I can't see it clearly; how it is used. Can you give an example? I understand that most of time, we don't need cluster ID; it is only needed when a snapshot operation is called. (* please put this in the comment *)